test/unit/math/binaryOps/src/binaryOps.cpp

```diff
@@ -0,0 +1,627 @@
+                                argX,
+                                argY
+                            );
+                    }
+                    break;
+                case Range::NOT_ZER0:
+                    for( TIdx i( 0 ); i < sizeArgs; ++i )
+                    {
+                        if( i == sizeArgs / 2 - 1 || i == sizeArgs / 2 )
+                            continue;
+                        argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx * sizeArgs] =
+                            structs[gridThreadIdx].alpakaOp(
+                                acc,
+                                argX,
+                                argY
+                            );
+                    }
+                    break;
+
+                case Range::X_NOT_ZERO:
+                    for( TIdx i( 0 ); i < sizeArgs; ++i )
+                    {
+                        if( i == sizeArgs / 2 - 1 || i == sizeArgs / 2 )
+                            argX = argsX[0];
+                        else
+                            argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx * sizeArgs] =
+                            structs[gridThreadIdx].alpakaOp(
+                                acc,
+                                argX,
+                                argY
+                            );
+                    }
+                    break;
+
+                case Range::Y_NOT_ZERO:
+                    for( TIdx i( 0 ); i < sizeArgs; ++i )
+                    {
+                        if( i == sizeArgs / 2 - 1 || i == sizeArgs / 2 )
+                            argY = argsY[0];
+                        else
+                            argY = argsY[i];
+                        argX = argsX[i];
+
+                        results[i + gridThreadIdx * sizeArgs] =
+                            structs[gridThreadIdx].alpakaOp(
+                                acc,
+                                argX,
+                                argY
+                            );
+                    }
+                    break;
+
+                case Range::UNRESTRICTED:
+                    for( TIdx i( 0 ); i < sizeArgs; ++i )
+                    {
+                        argX = argsX[i];
+                        argY = argsY[i];
+                        results[i + gridThreadIdx * sizeArgs] =
+                            structs[gridThreadIdx].alpakaOp(
+                                acc,
+                                argX,
+                                argY
+                            );
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+        }
+    }
+};
+
+
+struct TestTemplate
+{
+    template<
+        typename TAcc>
+    void operator()( )
+    {
+        using Data = double; // this can be changed to float without problems
```

Comment:
single + double precision have to be tested, so you could provide test cases with TestTemplate<float> and TestTemplate<double>